pkg/fleet/installer/service/apm_inject.go

```diff
@@ -212,3 +216,31 @@ func (a *apmInjectorInstaller) verifySharedLib(ctx context.Context, libPath stri
 	}
 	return nil
 }
+
+func shouldInstrumentHost(execEnvs *env.Env) bool {
+	switch execEnvs.InstallScript.APMInstrumentationEnabled {
+	case env.APMInstrumentationEnabledHost, env.APMInstrumentationEnabledAll, env.APMInstrumentationNotSet:
+		return true
+	case env.APMInstrumentationEnabledDocker:
+		return false
+	default:
+		log.Warnf("Unknown value for DD_APM_INSTRUMENTATION_ENABLED: %s. Supported values are all/docker/host", execEnvs.InstallScript.APMInstrumentationEnabled)
+		return false
+	}
+}
+
+func shouldInstrumentDocker(execEnvs *env.Env) bool {
+	switch execEnvs.InstallScript.APMInstrumentationEnabled {
+	case env.APMInstrumentationEnabledDocker, env.APMInstrumentationEnabledAll, env.APMInstrumentationNotSet:
+		return true
+	case env.APMInstrumentationEnabledHost:
+		return false
+	default:
+		log.Warnf("Unknown value for DD_APM_INSTRUMENTATION_ENABLED: %s. Supported values are all/docker/host", execEnvs.InstallScript.APMInstrumentationEnabled)
+		return false
+	}
+}
+
+func mustInstrumentDocker(execEnvs *env.Env) bool {
+	return execEnvs.InstallScript.APMInstrumentationEnabled == env.APMInstrumentationEnabledDocker
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏Comments would be nice as we won't have the nice markdown table in the code

Reply:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏Why not just pass the `APMInstrumentationEnabled` env var?

Reply:
I'm part of the code is comment club, the function name shouldInstrument/mustInstrument are not self-explanatory?

Reply:
When not taken in context, not really. Having the different cases summarised somewhere in the code would help those not familiar with the injection.

I won't block the PR on this.